README.md

```diff
@@ -70,6 +70,89 @@ task build
 task test:e2e
 ```
 
+## Deployment
+
+To deploy the Directory, you can use the provided `Taskfile` commands to start the necessary services and deploy the Directory server.
+
+### Local OCI Provider and Directory
+
+To start a local OCI registry server for storage and the Directory server Golang module, use the following commands:
+
+```bash
+task server:store:start
+task server:start
+```
+
+These commands will set up a local environment for development and testing purposes.
+
+### Helm chart
+
+To deploy the Directory using Helm charts, you can create a Kubernetes cluster using `kind` and then deploy the Directory Helm chart. Use the following commands:
+
+```bash
+task deploy:k8s:bootstrap
+task deploy:k8s:cleanup
+```
+
+The `bootstrap` task will create a Kubernetes cluster and deploy the Directory Helm chart, while the `cleanup` task will remove the cluster and clean up any resources created during the deployment.
+
+## CLI Commands
+
+The Directory CLI provides `build`, `push`, and `pull` commands to interact with the Directory server. Below are the details on how to run each command.
+
+To run these commands, you can either:
+* Use the binary compiled with `task cli:compile` like `./bin/dirctl <command> <args>`
+* Use the CLI module by navigating to the `cli` directory and running `go run cli.go <command> <args>`
+
+### Build Command
+
+The `build` command is used to compile and build the agent data model.
+
+Usage:
+```bash
+dirctl build [options] <path-to-agent>
+```
+
+Options:
+- `--name` : Name of the agent.
+- `--version` : Version of the agent.
+- `--artifact-url` : URL of the agent artifact.
+- `--artifact-type` : Type of the agent artifact.
+- `--author` : Authors to set for the agent. Overrides builder defaults. Example usage: `--author author1 --author author2`.
+- `--category` : Categories to set for the agent. Overrides builder defaults. Example usage: `--category category1 --category category2`.
+- `--created-at` : Agent creation time in RFC3339 format.
+- `--llmanalyzer` : Enable LLMAnalyzer extension.
+
+### Push Command
+
+The `push` command is used to publish the built agent data model to the store.
```

Comment:
we should also specify that push expects an input to be JSON formatted data for agent data models